### DISCUSSION TITLE
Does Tiny USB support two physcial host ports on the RP2040?

### DISCUSSION THREAD

**Discussion by morio** [hathach/tinyusb#1631] [Q&A]
Is it possible to use TinyUsb to control two physical USB host ports?
Specifically on the RP2040 I'd like to use the builtin USB as host and the Pico PIO USB also as a host port at the same time.
This would be to allow a USB keyboard in one port and a USB mouse in the other. 

I've seen the host_hid_to_device example in the Pico PIO USB project (https://github.com/sekigon-gonnoc/Pico-PIO-USB/tree/main/examples/host_hid_to_device_cdc) but that uses a device and a host.

Thanks!

**Comment by paulhamsh** [comment]
I think it should - the PIO library is just a way to implement USB in PIO and has a hcd file, so should provide a host.
So the question is can two hosts be run at the same time in TinyUSB - not sure about that, sorry

**Reply by hathach** [reply]
theoretically and technically, It could, but current implementation only pick one driver over other. Since in my initial implementation, I was mostly focusing on 1 host and 1 device which is the most common use case for rp2040. For now, please use an external hub. I will try to see if I could pull this off in the next release.

**Reply by hathach** [reply]
I create an issue as reminder #1673
